pkg/apiserver/rest/dualwriter_mode1.go

```diff
@@ -240,3 +264,7 @@ func (d *DualWriterMode1) ConvertToTable(ctx context.Context, object runtime.Obj
 func (d *DualWriterMode1) Compare(storageObj, legacyObj runtime.Object) bool {
 	return d.Storage.Compare(storageObj, legacyObj)
 }
+
+func (d *DualWriterMode1) waitPendingActions() {
```

Comment:
this new method seems never to be called, is that intentional?

Reply:
We use it in the tests when we call [waitPendingActions](https://github.com/grafana/grafana/pull/88886/files#diff-523751c29f56877dc6e5c815cc1e77cabae0247d3237f135aeb72212a65bac42R25)

Reply:
it might be cleaner to explicitly call `dw.pendingActions.Wait()` in the tests then... (though, see my concerns expressed in a separate comment about this whole scheme)

Reply:
I tried, but as the constructor returns a `DualWriter` interface I thought it would be impractical to type-assert the returned value in every test to make sure it's a `DualWriterMode1`. There are other three implementations of `DualWriter`, so I thought of using the small interface and type-asserting to it instead in case we also need to do the same in the other ones. I understand that this may not be obvious when looking at the code, so this definitely needs attention. Do you think it would be clearer if we: (1) type-assert to the specific implementation so we use the method; or (2) add a comment on the method implementation to state this? (or something else)